src/dynamics/ergodic/ergodic.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2022 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import dynamics.ergodic.measure_preserving
+
+/-!
+# Ergodic maps and measures
+
+Let `f : α → α` be measure preserving with respect to a measure `μ`. We say `f` is ergodic with
+respect to `μ` (or `μ` is ergodic with respect to `f`) if the only measurable sets `s` such that
+`f⁻¹' s = s` are either almost empty or full.
+
+In this file we define ergodic maps / measures together with quasi-ergodic maps / measures and
+provide some basic API.
+
+# Main definitions:
+
+ * `pre_ergodic`: the ergodicity condition without the measure preserving condition. This exists
+ to share code between the `ergodic` and `quasi_ergodic` definitions.
+ * `ergodic`: the definition of ergodic maps / measures.
+ * `quasi_ergodic`: the definition of quasi ergodic maps / measures.
+ * `ergodic.quasi_ergodic`: an ergodic map / measure is quasi ergodic.
+ * `quasi_ergodic.ae_empty_or_univ'`: when the map is quasi measure preserving, one may relax the
+ strict invariance condition to almost invariance in the ergodicity condition.
+
+-/
+
+open set function filter measure_theory measure_theory.measure
+
+variables {α : Type*} [measurable_space α] (f : α → α) {s : set α}
```

Comment:
Can you replace `[measurable_space α]` by an implicit argument `{m : measurable_space α}`? You will have to `include m` as well.
As written currently, when you apply a lemma the measurable space guessed by the typeclass might be different from the one from the measure, and your lemmas/defs won't work in that case. If you use an implicit argument, `m` will be taken from the measure and will always match.

Reply:
Good point, done.

I was aware of this design pattern in our measure theory but somehow forgot to follow it here. Thanks!